packages/sources/anchorage/src/transport/wallet.ts

```diff
@@ -0,0 +1,159 @@
+  requester!: Requester
+  settings!: WalletTransportTypes['Settings']
+
+  async initialize(
+    dependencies: TransportDependencies<WalletTransportTypes>,
+    adapterSettings: WalletTransportTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+    this.requester = dependencies.requester
+    this.settings = adapterSettings
+  }
+
+  async backgroundHandler(
+    context: EndpointContext<WalletTransportTypes>,
+    entries: RequestParams[],
+  ) {
+    await Promise.all(entries.map(async (param) => this.handleRequest(param)))
+    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+  }
+
+  async handleRequest(param: RequestParams) {
+    let response: AdapterResponse<WalletTransportTypes['Response']>
+    try {
+      response = await this._handleRequest(param)
+    } catch (e) {
+      const errorMessage = e instanceof Error ? e.message : 'Unknown error occurred'
+      logger.error(e, errorMessage)
+      response = {
+        statusCode: (e as AdapterError)?.statusCode || 502,
+        errorMessage,
+        timestamps: {
+          providerDataRequestedUnixMs: 0,
+          providerDataReceivedUnixMs: 0,
+          providerIndicatedTimeUnixMs: undefined,
+        },
+      }
+    }
+    await this.responseCache.write(this.name, [{ params: param, response }])
+  }
+
+  async _handleRequest(
+    params: RequestParams,
+  ): Promise<AdapterResponse<WalletTransportTypes['Response']>> {
+    const { vaultId, chainId, network } = params
+
+    const providerDataRequestedUnixMs = Date.now()
+
+    const wallets = await this.fetchWallets(vaultId)
+
+    const addresses = wallets.map((w) => {
+      return {
+        address: w.depositAddress.address,
+        chainId,
+        network,
+      }
+    })
+
+    return {
+      data: {
+        result: addresses,
+      },
+      statusCode: 200,
+      result: null,
+      timestamps: {
+        providerDataRequestedUnixMs,
+        providerDataReceivedUnixMs: Date.now(),
+        providerIndicatedTimeUnixMs: undefined,
+      },
+    }
+  }
+
+  async fetchWallets(vaultId: string) {
+    const wallets = []
+    let hasNext = true
+    const requestConfig = {
+      baseURL: this.settings.API_ENDPOINT,
+      url: `/vaults/${vaultId}/wallets?limit=${this.settings.API_LIMIT}`,
+      headers: {
+        'Api-Access-Key': this.settings.API_KEY,
+      },
+    }
+
+    while (hasNext) {
+      const reqKey = requestConfig.baseURL + requestConfig.url
+      const response = await this.requester.request<WalletResponse>(reqKey, requestConfig)
+      wallets.push(...response.response.data.data)
+      hasNext = response.response.data.page.next !== null
+      if (response.response.data.page.next) {
+        // Remove the '/v2' prefix from the URL as it's already part of the baseURL
+        requestConfig.url = response.response.data.page.next.replace('/v2', '')
```

Comment:
why don't we just get rid of it from the base API then, and add it above to the `requestConfig.url`?